### PR TITLE
Remove Cardinality

### PR DESCRIPTION
**Pull request by adamb3ard**
Description
----
Cardinality is a mix of different things right now on our documentation. Sometimes it's the base spec, and sometimes it's updated to show we require the field (eg: 1..). I went through and removed the cardinality column from the body field tables since we already link to the base spec. This seemed to me the only place using the cardinality field in the YAML files, so I removed those fields as well. I also went through and mentioned that the field is a list of the specific datatype where applicable. I also updated the notes on some fields where our cardinality did not match the spec.
Bonus: Removing the column left a bit of space, so I added some CSS to center values in the body field tables.

Here is an example of a table without cardinality. The changing of adding "List of" is also visible here.
bigSpaceLeft

Here is the table with my proposed CSS update:
bigSpaceCenter

Patch also had cardinality fields, so I removed that. Here is a demo of that change:
PatchDemo

DSTU2 DocumentReference and Appointment required a note to specify a max cardinality that doesn't match the spec:
DocRefAuthorNote
AppointmentSlotNote

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

File: lib/resources/dstu2/appointment.yaml [hunk 1]
```diff
@@ -47,9 +45,9 @@ fields:
 
 - name: slot
   required: 'Yes'
-  cardinality: 1..1
   type: Reference (Slot)
```
Comment (MaxPhilips) [inline lib/resources/dstu2/appointment.yaml:48]: List of Reference (Slot)

File: lib/resources/dstu2/document_reference.yaml [hunk 2]
```diff
@@ -75,7 +72,6 @@ fields:
       info_link: /millennium/dstu2/proprietary-codes/#code-set-72-event-code
 - name: author
   required: 'No'
-  cardinality: 0..1
   type: Reference (Practitioner)
```
Comment (MaxPhilips) [inline lib/resources/dstu2/document_reference.yaml:75]: List of Reference (Practitioner)

File: lib/resources/dstu2/document_reference.yaml [hunk 3]
```diff
@@ -86,11 +82,12 @@ fields:
         }
       ]
     }
-  note: If author is not provided, the author will be determined based on the access token.
+  note: At most one practitioner can be populated.
```
Comment (MaxPhilips) [inline lib/resources/dstu2/document_reference.yaml:85]: prefer to implement multiple notes via an unordered list. reference: https://github.com/cerner/fhir.cerner.com/blob/881ebafa32dad7f57c4985f6799edb46ec8e81aa/lib/resources/r4/patient_patch.yaml#L33

File: lib/resources/dstu2/medication_order.yaml [hunk 4]
```diff
@@ -24,7 +23,6 @@ fields:
 
 - name: medicationCodeableConcept
   required: 'No'
```
Comment (MaxPhilips) [inline lib/resources/dstu2/medication_order.yaml:25]: an old cardinality of 1..1 does not mesh with this field being un-required. can you double check our implementation?

File: lib/resources/dstu2/medication_statement.yaml [hunk 5]
```diff
@@ -52,7 +48,6 @@ fields:
 
 - name: reasonForUse[x]
   required: 'no'
```
Comment (MaxPhilips) [inline lib/resources/dstu2/medication_statement.yaml:50]: capitalize any No's that aren't already

File: lib/resources/dstu2/medication_statement.yaml [hunk 6]
```diff
@@ -174,7 +161,7 @@ fields:
               "start": "2014-11-03T14:38:00.000-05:00"
             }
           }
-          
+
           {
```
Comment (MaxPhilips) [inline lib/resources/dstu2/medication_statement.yaml:165]: can you split these into multi-examples? reference: https://github.com/cerner/fhir.cerner.com/blob/6bbed7b93178cdd00866c5abd685770a48a58fd3/lib/resources/r4/encounter.yaml#L65

File: lib/resources/dstu2/medication_statement.yaml [hunk 7]
```diff
@@ -226,7 +211,6 @@ fields:
 
   - name: asNeeded[x]
     required: 'No'
-    cardinality: 0..1
     type: boolean | CodeableConcept
     description: Whether the Medication is only taken when needed within a specific dosing schedule.
     example: |
```
Comment (MaxPhilips) [inline lib/resources/dstu2/medication_statement.yaml:216]: multi-examples

File: lib/resources/r4/allergy_intolerance.yaml [hunk 8]
```diff
@@ -311,7 +296,6 @@ fields:
 
 - name: onset_x_
```
Comment (MaxPhilips) [inline lib/resources/r4/allergy_intolerance.yaml:297]: can you check quickly if there's a reason we aren't using square brackets here? might need to escape the characters

File: lib/resources/r4/appointment.yaml [hunk 9]
```diff
@@ -103,7 +96,6 @@ fields:
 
 - name: serviceType
   required: 'Yes'
```
Comment (MaxPhilips) [inline lib/resources/r4/appointment.yaml:98]: previous cardinality 0..1 doesn't match with required field, can you double check?

File: lib/resources/r4/condition.yaml [hunk 10]
```diff
@@ -29,7 +27,6 @@ fields:
 
 - name: clinicalStatus
   required: 'Yes'
```
Comment (MaxPhilips) [inline lib/resources/r4/condition.yaml:29]: double check both instances of clinicalStatus here, and category and code below

File: lib/resources/r4/coverage.yaml [hunk 11]
```diff
@@ -29,7 +27,6 @@ fields:
 
 - name: subscriber
   required: 'Yes'
```
Comment (MaxPhilips) [inline lib/resources/r4/coverage.yaml:29]: double check subscriber and relationship

File: lib/resources/r4/device.yaml [hunk 12]
```diff
@@ -25,7 +23,6 @@ fields:
 
 - name: type
   required: 'Yes'
```
Comment (MaxPhilips) [inline lib/resources/r4/device.yaml:25]: double check

File: lib/resources/r4/encounter.yaml [hunk 13]
```diff
@@ -46,7 +44,6 @@ fields:
 
   - name: type
     required: 'Yes'
```
Comment (MaxPhilips) [inline lib/resources/r4/encounter.yaml:46]: double check type, system, value, class, the next three instances of type, and individual

File: lib/resources/r4/financial_transaction.yaml [hunk 14]
```diff
@@ -17,8 +16,7 @@ fields:
 
 - name: identifier
   required: 'Yes'
-  cardinality: 0..*
-  type: Identifier
+  type: List of Identifier
```
Comment (MaxPhilips) [inline lib/resources/r4/financial_transaction.yaml:19]: double check here, and most extensions

File: lib/resources/r4/immunization.yaml [hunk 15]
```diff
@@ -127,7 +121,6 @@ fields:
 
 - name: encounter
   required: 'Yes'
```
Comment (MaxPhilips) [inline lib/resources/r4/immunization.yaml:123]: double check

File: lib/resources/r4/allergy_intolerance.yaml (hunk 8 above)
Comment (MaxPhilips) [reply lib/resources/r4/allergy_intolerance.yaml:297]: based on other files plain old unescaped square brackets seem to work, at least in a terminology table

File: lib/resources/r4/organization.yaml [hunk 16]
```diff
@@ -104,7 +98,6 @@ fields:
 
   - name: name
     required: 'Yes'
```
Comment (MaxPhilips) [inline lib/resources/r4/organization.yaml:100]: double check

File: lib/resources/r4/patient.yaml [hunk 17]
```diff
@@ -81,15 +76,13 @@ fields:
...
   required: 'No'
-  cardinality: 0..*
-  type: BackboneElement
+  type: List of BackboneElement
   description: Language which may be used to communicate with the patient about his or her health.
   action: terminology
   children:
 
   - name: language
     required: 'No'
```
Comment (MaxPhilips) [inline lib/resources/r4/patient.yaml:85]: double check here and identifier

File: lib/resources/r4/procedure.yaml [hunk 18]
```diff
@@ -43,7 +42,7 @@ fields:
 
 - name: code
   required: 'Yes'
-  cardinality: 0..1
+  required: 'No'
```
Comment (MaxPhilips) [inline lib/resources/r4/procedure.yaml:45]: duped required keys

File: static/css/documentation.css [hunk 19]
```diff
@@ -203,6 +203,10 @@ a img {
   text-decoration: none;
 }
 
+.bodyField {
```
Comment (MaxPhilips) [inline static/css/documentation.css:206]: we tend not to use camel case classes in this project. how about body-field?

File: lib/resources/dstu2/appointment.yaml (hunk 1 above)
Comment (adamb3ard) [reply lib/resources/dstu2/appointment.yaml]: [Updated](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-dda110f273dfe3a3ddac84fe6dacc327L48)

File: lib/resources/dstu2/document_reference.yaml (hunk 2 above)
Comment (adamb3ard) [reply lib/resources/dstu2/document_reference.yaml]: [updated](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-eefc10b8b4a495f72e678cead5851448L75)

File: lib/resources/dstu2/document_reference.yaml (hunk 3 above)
Comment (adamb3ard) [reply lib/resources/dstu2/document_reference.yaml]: [listed](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-eefc10b8b4a495f72e678cead5851448L85)

File: lib/resources/dstu2/medication_order.yaml (hunk 4 above)
Comment (adamb3ard) [reply lib/resources/dstu2/medication_order.yaml]: [double checked](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-e19db23d6092046f31617ecce00ba335L25)

File: lib/resources/dstu2/medication_statement.yaml (hunk 5 above)
Comment (adamb3ard) [reply lib/resources/dstu2/medication_statement.yaml]: [Fixed that, no cap](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-5321e3dd849f4c21936b5a38c0bf2092L50)

File: lib/resources/dstu2/medication_statement.yaml (hunk 6 above)
Comment (adamb3ard) [reply lib/resources/dstu2/medication_statement.yaml:166]: [split into a second example](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-5321e3dd849f4c21936b5a38c0bf2092R165)

File: lib/resources/dstu2/medication_statement.yaml (hunk 7 above)
Comment (adamb3ard) [reply lib/resources/dstu2/medication_statement.yaml:217]: [Split into 2 example](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-5321e3dd849f4c21936b5a38c0bf2092R230)

File: lib/resources/r4/allergy_intolerance.yaml (hunk 8 above)
Comment (adamb3ard) [reply lib/resources/r4/allergy_intolerance.yaml]: [I squared up.](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-eeda3e608dc0db5d6d7c877900383353L297) I also added square brackets [here](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-abb99d2139d79f8a2a8f1d214f5f6882L301) and [here.](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-abb99d2139d79f8a2a8f1d214f5f6882L316)

File: lib/resources/r4/appointment.yaml (hunk 9 above)
Comment (adamb3ard) [reply lib/resources/r4/appointment.yaml]: [double checked](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-fbb215a73c336ca1ac34d91de2fb411eL98)

File: lib/resources/r4/condition.yaml (hunk 10 above)
Comment (adamb3ard) [reply lib/resources/r4/condition.yaml:29]: Through a combination of making calls and reading the validators, I have came to the conclusion that the 'required' field is correct on these. However for this clinicalStatus the action is set to a terminology, so having it be required kinda makes sense, even though I don't think that does anything.

File: lib/resources/r4/coverage.yaml (hunk 11 above)
Comment (adamb3ard) [reply lib/resources/r4/coverage.yaml:29]: [Double checked.](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-af9c966415db83d073f1bcec5ed6f875L73) I believe subscriber is required, but relationship is not required.

File: lib/resources/r4/device.yaml (hunk 12 above)
Comment (adamb3ard) [reply lib/resources/r4/device.yaml]: [Double checked](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-7bad88b7f4265915d4ac88a059a5d185L25)

File: lib/resources/r4/encounter.yaml (hunk 13 above)
Comment (adamb3ard) [reply lib/resources/r4/encounter.yaml:46]: [Double checked](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-876b1ef3d3fb228190dea594ca0a12abR73) and added notes to clarify sub-fields are required when an optional field is given. [I also past tense'd a require](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-876b1ef3d3fb228190dea594ca0a12abR337)

File: lib/resources/r4/financial_transaction.yaml (hunk 14 above)
Comment (adamb3ard) [reply lib/resources/r4/financial_transaction.yaml:19]: After reading the notes provided on the fields and checking verifies, I think these are correct.

File: lib/resources/r4/immunization.yaml (hunk 15 above)
Comment (adamb3ard) [reply lib/resources/r4/immunization.yaml]: [Double checked](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-2c50e8f41daaba3a640861e7b6e81488L123)

File: lib/resources/r4/organization.yaml (hunk 16 above)
Comment (adamb3ard) [reply lib/resources/r4/organization.yaml:100]: Based on the note and verifiers I believe this is required.

File: lib/resources/r4/patient.yaml (hunk 17 above)
Comment (adamb3ard) [reply lib/resources/r4/patient.yaml]: [I changed location](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-07c686ec4248977e084ba5a602c761ceL85), but I think identifier is correct.

File: lib/resources/r4/procedure.yaml (hunk 18 above)
Comment (adamb3ard) [reply lib/resources/r4/procedure.yaml]: [Removed the dupe and some missed cardinalities](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.yaml#diff-07c686ec4248977e084ba5a602c761ceL85)

File: static/css/documentation.css (hunk 19 above)
Comment (adamb3ard) [reply static/css/documentation.css]: [un-camel'd](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.css#diff-331344c9aef74ed7c8415f3632a7c619L206) and updated the [def table](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.erb#diff-1adfd305d8a0445e782d6f126c58d760L12) and [patch def table](https://github.com/cerner/fhir.cerner.com/pull/374/commits/6fb0b4a5a1bdcd0b07e086f542d44ded83b12430?file-filters%5B%5D=.erb#diff-bac15500a3095fd0a82601de2a06101eL12)

File: lib/resources/r4/condition.yaml (hunk 10 above)
Comment (MaxPhilips) [reply lib/resources/r4/condition.yaml:29]: i think we should remove the key entirely in these cases if it is not used

Comment (adamb3ard) [reply lib/resources/r4/condition.yaml]: [Removed required field from terminologies](https://github.com/cerner/fhir.cerner.com/pull/374/commits/b2c52fced44727f05e6d5b7bafd68cbbb969ced7)

**Comment by FFSol** [comment]
> most likely need to incorporate some sort of required field into the patch definition table as a replacement for cardinality

Why expose whether it's required if the PATCH operation has no implicitly required fields? A PUT would have requirements, but the PATCH is partial so nothing is really 'required'

**Comment by MaxPhilips** [comment]
@FFSol true. i guess i was thinking about cases where certain patch operations are required together, which should be covered by implementation notes

**Comment by FFSol** [comment]
@MaxPhilips good point. I wasn't aware of required joint patch operations; we can discuss adding implementation notes to fill that gap maybe. That would be outside the scope of this change, though so +1

**Comment by wseymour15** [comment]
Cardinality has been removed from DSTU2 resources

Screen Shot 2020-05-22 at 11 10 42 AM

Cardinality has been removed from R4 Resources

Screen Shot 2020-05-22 at 11 11 23 AM

**Pull request merged**